src/main/java/software/amazon/awssdk/crt/s3/S3Client.java

```diff
@@ -122,6 +122,12 @@ private void onShutdownComplete() {
 
     public S3MetaRequest makeMetaRequest(S3MetaRequestOptions options) {
 
+        if(isNull()) {
+            Log.log(Log.LogLevel.Error, Log.LogSubject.S3Client,
+                    "S3Client.makeMetaRequest has invalid client. The client can not be used after it is closed.");
+            return null;
```

Comment:
sins of the past: why is this function returning null instead of throwing exceptions?

is this C 💁‍♂️🦋 ?

Reply:
Thanks, fixed.